index.rst

```diff
@@ -1,61 +1,643 @@
+subimage from the same DECam image (`Figure 4b <#figure-4b>`__).
+
+.. figure:: _static/fig04a.png
+   :name: figure-4a
+
+   Subsections of a DECam Zogy image subtraction, including warped and
+   PSF-matched template, science image, and the results of the
+   "standard" and image-space versions of the Zogy algorithm.
+
+.. figure:: _static/fig04b.png
+   :name: figure-4b
+
+   Subsections of the same DECam Zogy image subtraction as in Figure 4a.
+
+I should note that this fringing was observed by Tim Axelrod in another
+Zogy implementation when a certain PSFex PSF configuration was used
+(pixel based? too small PSF dimensions? "It certainly is a result of bad
+parameters to psfex, and in particular the footprint size for
+determining the psf being way too big for this data."). I include his
+example below in `Figure 5 <#figure-5>`__. It appears to be an
+:math:`S_{corr}` image (see Section 2.3, below).
+
+.. figure:: _static/fig05.png
+   :name: figure-5
+
+   Example Zogy image with fringing from Tim Axelrod
+
+**Timing:** The current implementation of Zogy takes roughly 26.6
+seconds, or :math:`0.78\times` as long (i.e., is :math:`\sim22\%`
+faster) to run than the AL algorithm with decorrelation enabled. There
+has been limited attempt to date to optimize the Zogy algorithm, and
+some simple profiling is likely to highlight several bottlenecks.
+
+**Additional known issue:** Zogy relies upon FFTs of the PSFs of both
+input images. If those PSFs are not the same dimension, then one of them
+needs to be padded or trimmed. We also need to ensure that each PSFs are
+centered correctly, and centered at the same pixel coordinate. There is
+much code in ``lsst.ip.diffim.zogy`` for making these corrections, yet
+sometimes the resulting Zogy diffim has 1-pixel offsets from expected. I
+have not yet been able to fix this in all cases, and it is not clear why
+for some images this becomes an issue, while for others it is not.
+
+2.1. Variants (image-space convolutions)
+----------------------------------------
+
+The convolutions in Zogy may be performed in image (real)-space rather
+than in frequency space. This is beneficial in the LSST stack as then
+the convolutions may be performed using the ``afw`` framework, which
+accounts for masks and propagates the convolutions through to the
+variance and mask planes of the exposures. The image-space convolution
+Zogy images are shown in the bottom-right of `Figure 4 <#figure-4>`__
+and `Figure 5 <#figure-5>`__. Because the convolution kernel is
+spatially constrained, we see that the artifacts evident in the
+"standard" Zogy implementation (bottom left of those figures) are also
+spatially constrained. However, it is also evident that echo-like
+artifacts are also generated in the image-space version which can be
+severe surrounding the brightest stars. These artifacts lead to a
+greater number of false positive detections (472 vs. 257 before merging;
```

Comment:
what does "merging" refer to here?

Reply:
positive and negative sources in the diffim are merged (into dipoles etc).